Tests/SonarScanner.MSBuild.Common.Test/MsBuildPathSettingsTests.cs

```diff
@@ -270,14 +290,36 @@ public void GetGlobalTargetsPaths_WhenProgramFilesNotEmpty_ReturnsExpectedPaths(
                 "C:\\Program\\MSBuild\\15.0\\Microsoft.Common.Targets\\ImportBefore");
         }
 
-        private static Func<bool> IsWindows(bool result) => () => result;
-        private static Func<string, bool> DirExists(bool result) => path => result;
-        private static Func<Environment.SpecialFolder, Environment.SpecialFolderOption, string> GetFolderPath(params (Environment.SpecialFolder, string)[] paths) =>
-            (folder, option) =>
-            {
-                // Bug #681 - the Create option doesn't work on some NET Core versions on Linux
-                option.Should().NotBe(Environment.SpecialFolderOption.Create);
-                return paths.First(p => p.Item1 == folder).Item2;
-            };
+        private MsBuildPathSettings MsBuildPathSettings(
+            (Environment.SpecialFolder, string)[] paths,
+            bool isWindows,
+            Func<string, bool> directoryExistsFunc,
+            bool isMacOSX) =>
```

Comment:
`isWindows`, `isMacOSX`, and `isLinux` would be mutual exclusive. The tests would be more stable and easier to read if this is an enum instead:
```cs
enum OS
{
    Windows,
    Linux,
    MacOSX,
}
```